main.go

```diff
@@ -0,0 +1,121 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"strings"
+)
+
+// TODO: add language flag diversion, unit unit test
+
+func main() {
+	// peeper subcommand "get"
+	getCmd := flag.NewFlagSet("get", flag.ExitOnError)
+	// Flags for "get" subcommand
+	// ("flag name", default, "Flag explanation")
+	getAll := getCmd.Bool("all", false, "Get full shortcut list")
+	getByKeyword := getCmd.String("keyword", "", "Find a shortcut with keyword ")
+
+	// peeper subcommand "add"
+	addCmd := flag.NewFlagSet("add", flag.ExitOnError)
+	// Flags(inputs) for "add" subcommand
+	addName := addCmd.String("name", "", "Name of the shortcut")
+	addShortcut := addCmd.String("shortcut", "", "Shortcut")
+
+	// peeper subcommand "delete"
+	deleteCmd := flag.NewFlagSet("delete", flag.ExitOnError)
+	// Flags for "delete" subcommand
+	deleteAll := deleteCmd.Bool("all", false, "Delete whole shortcut list")
+	deleteByName := deleteCmd.String("name", "", "Delete a shortcut by name")
+
+	if len(os.Args) < 2 {
+		fmt.Println("Exepected either 'get', 'add', 'delete' subcommands")
+	}
+
+	switch os.Args[1] {
+	case "get":
+		HandleGet(getCmd, getAll, getByKeyword)
+	case "add":
+		HandleAdd(addCmd, addName, addShortcut)
+	case "delete":
+		HandleDelete(deleteCmd, deleteAll, deleteByName)
+		// TODO: Do we need to add default and return error for the case of the command doesn't exist?
+	}
+}
+
+func HandleGet(getCmd *flag.FlagSet, all *bool, keyword *string) {
+	getCmd.Parse(os.Args[2:])
+
+	if *all == false && *keyword == "" {
```

Comment:
Are these de-references safe, or is there a chance that this could panic? (They might be safe, I can't remember)

Reply:
>de-references safe

Was not aware of this point, thanks, will update this part later.